src/openrct2/ride/Vehicle.cpp

```diff
@@ -3133,11 +3132,12 @@ static bool vehicle_current_tower_element_is_top(Vehicle* vehicle)
 static void vehicle_update_travelling_boat_hire_setup(Vehicle* vehicle)
 {
     vehicle->var_34 = vehicle->sprite_direction;
-    vehicle->track_x = vehicle->x & 0xFFE0;
-    vehicle->track_y = vehicle->y & 0xFFE0;
+    vehicle->TrackLocation.x = vehicle->x;
+    vehicle->TrackLocation.y = vehicle->y;
+    vehicle->TrackLocation.ToTileStart();
```

Comment:
This function returns instead of updating the object, so it should be:
```suggestion
    vehicle->TrackLocation = vehicle->TrackLocation.ToTileStart();
```